doc/dev/kclient_dev_bootstrap.rst

```diff
@@ -0,0 +1,474 @@
+
+Configure the Kernel
+^^^^^^^^^^^^^^^^^^^^
+
+From within the **guest VM**, copy the current kernel configuration file from
+``/boot/config-$(uname -r)`` to ``.config`` in the kernel source directory.
+This ensures that the new build will have a configuration similar to the
+currently running guest kernel.
+
+.. code-block:: bash
+
+   cp /boot/config-$(uname -r) /path/to/linux/.config
+   make localmodconfig
+   make olddefconfig
+
+- ``make localmodconfig``: Updates ``.config`` to include only the modules
+  currently loaded on the host.
+- ``make olddefconfig``: Fills in any missing configuration options in
+  ``.config`` with default values.
+
+For more information on this process, refer to `the localmodconfig
+documentation at
+<https://www.kernel.org/doc/Documentation/admin-guide/README.rst>`_. This
+command will ensure that all necessary modules are configured.
+
+.. note:: You can alternatively use the `Ceph Kernel QA Config`_ for building
+   the kernel.
+
+We now have a kernel config with reasonable defaults for the architecture
+you're building on. The next thing to do is to enable configs which will build
+Ceph and provide the functionality needed to do testing.
+
+.. code-block:: bash
+
+    cat > ~/.ceph.config <<EOF
+    CONFIG_CEPH_LIB=m
+    CONFIG_CEPH_FS=m
+    CONFIG_CEPH_FSCACHE=y
+    CONFIG_CEPH_FS_POSIX_ACL=y
+    CONFIG_CEPH_FS_SECURITY_LABEL=y
+    CONFIG_CEPH_LIB_PRETTYDEBUG=y
+    CONFIG_DYNAMIC_DEBUG=y
+    CONFIG_DYNAMIC_DEBUG_CORE=y
+    CONFIG_FRAME_POINTER=y
+    CONFIG_FSCACHE=y
+    CONFIG_FSCACHE_STATS=y
+    CONFIG_FS_ENCRYPTION=y
+    CONFIG_FS_ENCRYPTION_ALGS=y
+    CONFIG_KGDB=y
+    CONFIG_KGDB_SERIAL_CONSOLE=y
+    CONFIG_XFS_FS=y
+    EOF
+
+In addition to enabling Ceph-related configs, we are also enabling some useful
+debug configs and XFS (as an alternative to ext4 if needed for our root file
+system).
+
+Merge the configs:
+
+
+.. code-block:: bash 
+
+
+   scripts/kconfig/merge_config.sh .config ~/.ceph.config
+
+
+Build the Kernel
+^^^^^^^^^^^^^^^^
+
+Compile the kernel using one of the following commands. Adjust the ``-j`` value
+as needed to match the number of cores available on the host for parallel
+processing.
+
+.. code-block:: bash
+
+   make
+
+or
+
+.. code-block:: bash
+
+   make -j16 LLVM=1 LLVM_IAS=1 CC='ccache clang' -s ARCH=x86_64 all compile_commands.json
+
+- ``LLVM=1``: Enables the use of the LLVM toolchain.
+- ``LLVM_IAS=1``: Enables the LLVM Integrated Assembler.
+- ``CC='ccache clang'``: Uses ``ccache`` to speed up compilation with
+  ``clang``.
+- ``ARCH=x86_64``: Specifies the target architecture for the build.
+- ``all compile_commands.json``: Builds the kernel and generates a
+  ``compile_commands.json`` file for tools that use this format.
+
+Install Modules on the Guest
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After the build completes, return to the guest and install the compiled modules
+and kernel:
+
+.. prompt:: bash $
+
+   sudo make modules_install && sudo make install
```

Comment:
For testing purposes, do we really want to advise having modules at all? I was advising against having any modules in the `kclient.rst` doc. This is to specifically avoid having to do `make install`.

Reply:
Ah, in this doc you're using the root disk to for booting the kernel.

Reply:
Yes, you have a base image that you can clone and have multiple version for example. 
It provides greater flexibility.